File: 7_Kyu_Kata/drone_fly_by/TypeScript/droneFlyBy.ts
// 7 kyu - Drone Fly-By  [ ID: 58356a94f8358058f30004b5  (drone-fly-by) ]
// URL: https://www.codewars.com/kata/58356a94f8358058f30004b5
// Category: undefined  |  Tags: FUNDAMENTALS | STRINGS | ARRAYS
// *****************************************************************************
export function flyBy(l: string, d: string): string {
    return l.replace(RegExp("x{" + Math.min(l.length, d.length) + "}"), "o".repeat(Math.min(l.length, d.length)))
  }

File: 7_Kyu_Kata/drone_fly_by/Python/drone_fly_by.py
# 7 kyu - Drone Fly-By  [ ID: 58356a94f8358058f30004b5  (drone-fly-by) ]
# URL: https://www.codewars.com/kata/58356a94f8358058f30004b5
# Category: NONE  |  Tags: FUNDAMENTALS | STRINGS | ARRAYS
# ******************************************************************************
import re
  
  def fly_by(lamps, drone):
      return re.sub(r"^x{1," + f"{(x:=len(drone))}" + r"}", "o" * (len(lamps) if len(lamps) < x else x), lamps)
  
